x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/job/DateHistogramGroupConfig.java

```diff
@@ -203,7 +320,7 @@ public boolean equals(final Object other) {
         if (this == other) {
             return true;
         }
-        if (other == null || getClass() != other.getClass()) {
+        if (other == null || other instanceof DateHistogramGroupConfig == false) {
```

Comment:
This one is a bit tricky.  This is there to accomodate the legacy option.  E.g. `5m` in `DateHistogramGroupConfig` == `5m` in `DateHistogramGroupConfig.FixedInterval`.

I could make the check more explicit and make sure it is either the same class or the super class, which would rule out `Fixed` vs `Calendar` comparisons?

Reply:
> I could make the check more explicit and make sure it is either the same class or the super class, which would rule out Fixed vs Calendar comparisons?

I think we should do this at a minimum as I think its important that when the legacy code is removed we don't have a situation where `Fixed == Calendar` is possible.

I'm wondering though when the comparison between legacy and new is actually needed? Presumably within a node/shard its either going to be interpreted as legacy OR as a new interval but never both? I suppose one place this could break is in the reduce on the coordinating node but do we actually need to do a comparison there?

Reply:
Thinking about it, I don't think this will ever actually happen in practice.  New nodes will always "upconvert" the old object into a new one (using that "from unknown interval") method.  I don't quite recall but this may have been added because of failing randomized serialization tests or something.  I'll investigate if we actually need it.

If there is a compelling reason for it, I'll add the `Fixed != Calendar` check at the very least.

Reply:
Yeah dunno, reverted this change and all tests pass.  

This is the rollup datehisto config object, and I don't think it's actually compared anywhere anyway so this leniency isn't needed anywhere from what i can see.  Think it might just have been overthinking the change when I first did it :)  Good catch, thanks!